numpy/ma/extras.py

```diff
@@ -1075,7 +1075,7 @@ def unique(ar1, return_index=False, return_inverse=False):
     numpy.unique : Equivalent function for ndarrays.
 
     """
-    output = np.unique(ar1,
+    output = np.unique(ar1.data[~ar1.mask],
                        return_index=return_index,
                        return_inverse=return_inverse)
```

Comment:
I don't think this will do the right thing when asked to return indices

Reply:
Thanks for the review @eric-wieser. Indeed, the problem turned out to be way deeper.